### ISSUE TITLE
Could you provide the code for SuperPoint?

### ISSUE THREAD

**Issue by shijiangming1** [djiajunustc/3D-LLaVA#7]
Thank you for your excellent work! The code of SuperPoint isn't publicly available yet, I'm writing to request access to the implementation for my research purposes.

**Comment by djiajunustc** [comment]
I simply followed https://github.com/Karbo123/segmentator.

**Comment by shijiangming1** [comment]
It seems that segmentator requires mesh input to function. Additionally, the paper seems to mention Nr3D, but I couldn't find its experimental results?

**Comment by shijiangming1** [comment]
import trimesh
import numpy as np
from segmentator import segment_mesh, segment_point, compute_vn

MESH_PATH = ??? # here is the path to your triangle mesh
mesh = trimesh.load_mesh(MESH_PATH)

# segment on mesh
vertices = torch.from_numpy(mesh.vertices.astype(np.float32))
faces = torch.from_numpy(mesh.faces.astype(np.int64))
ind = segment_mesh(vertices, faces) 
color_table = torch.randint(0, 256, size=(1 + ind.max(), 3))
np.savetxt("result_pc_mesh.txt", torch.cat([vertices, color_table[ind]], dim=1).numpy())

# segment on pointcloud
normals = torch.from_numpy(compute_vn(mesh).astype(np.float32))
edges = torch.from_numpy(mesh.edges.astype(np.int64)) # NOTE the edges can actually be obtained from knn graph or radius graph
ind = segment_point(vertices, normals, edges) 
np.savetxt("result_pc_point.txt", torch.cat([vertices, color_table[ind]], dim=1).numpy())

I simply followed https://github.com/Karbo123/segmentator, but the code doesn't seem to get and you provide superpoint_mask.

**Comment by djiajunustc** [comment]
It seems that segmentator requires mesh input to function. 
Please refer to the example of the link I shared.

Additionally, the paper seems to mention Nr3D, but I couldn't find its experimental results?
The training set of Nr3D is included in our training data.

**Comment by Jamin-Yang** [comment]
> It seems that segmentator requires mesh input to function. Additionally, the paper seems to mention Nr3D, but I couldn't find its experimental results?

Hi, how did you solve the problem?
